src/plugins/Dropaccelerator.cpp

```diff
@@ -0,0 +1,190 @@
+/*
+ * Dropaccelerator.h
+ *
+ *  Created on: 26 June 2020
+ *      Author: Koch
+ */
+
+#include "Dropaccelerator.h"
+
+//! @brief will be called to read configuration
+//!
+//!
+//!
+//! \param xmlconfig  read from config.xml
+void Dropaccelerator::readXML(XMLfileUnits& xmlconfig){
+
+    xmlconfig.getNodeValue("xposition", _xPosition);
+    xmlconfig.getNodeValue("yposition", _yPosition);
+    xmlconfig.getNodeValue("zposition", _zPosition);
+    xmlconfig.getNodeValue("dropradius", _Dropradius);
+    xmlconfig.getNodeValue("velocity", _Veloc);
+    xmlconfig.getNodeValue("starttime", _StartSimstep);
+    xmlconfig.getNodeValue("steps", _Steps);
+   
+
+	// SANITY CHECK
+    if(_INTERVAL < 1 || _Steps <= 0 || _Veloc < 0 || _StartSimstep < 0 || _xPosition <= 0. || _yPosition <= 0. || _zPosition <= 0. || _Dropradius <= 0)
+    {
+        global_log -> error() << "[Dropaccelerator] INVALID CONFIGURATION!!! DISABLED!" << std::endl;
+        global_log -> error() << "[Dropaccelerator] HALTING SIMULATION" << std::endl;
+        _enabled = false;
+        // HALT SIM
+        Simulation::exit(1);
+        return;
+    }
+
+	global_log -> info() << "[Dropaccelerator] settings:" << std::endl;
+    global_log -> info() << "                  xposition: " << _xPosition << std::endl;
+    global_log -> info() << "                  yposition: " << _yPosition << std::endl;
+    global_log -> info() << "                  zposition: " << _zPosition << std::endl;
+	global_log -> info() << "                  dropradius: " << _Dropradius << std::endl;
+    global_log -> info() << "                  velocity: " << _Veloc << std::endl;
+    global_log->info() << "                  starttime: " << _StartSimstep << std::endl;
+    global_log->info() << "                  steps: " << _Steps << std::endl;
+    
+}
+
+void Dropaccelerator::beforeForces(ParticleContainer* particleContainer,
+    DomainDecompBase* domainDecomp,
+    unsigned long simstep) {
+
+    double corrVeloc = -(_Veloc/_Steps);
```

Comment:
why `-`? can't you add this to the input file? that way everything would be cleaner.

Reply:
Yes I guess we'll add this to the input. I will be discussing all of these plugins in detail with Fabio later today, since there are obviously still some general problems with functionality,